### PR TITLE
Extend AnalyzerTest to support tests for DiagnosticSuppressor analyzers

### PR DESCRIPTION
**Pull request by tom-englert**
This implements scenario 1 from #915  by conditionally selecting dependencies and extend the analyzer validation code.

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Testing/Directory.Build.props [hunk 1]
```diff
@@ -25,6 +25,23 @@
...
+      <PropertyGroup>
+        <Testing_MicrosoftCodeAnalysisWorkspaces_Version>1.0.1</Testing_MicrosoftCodeAnalysisWorkspaces_Version>
+        <Testing_MicrosoftCodeAnalysisAnalyzers_Version>2.6.1</Testing_MicrosoftCodeAnalysisAnalyzers_Version>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <Testing_MicrosoftCodeAnalysisWorkspaces_Version>3.8.0</Testing_MicrosoftCodeAnalysisWorkspaces_Version>
+        <Testing_MicrosoftCodeAnalysisAnalyzers_Version>3.3.2</Testing_MicrosoftCodeAnalysisAnalyzers_Version>
+        <DefineConstants Condition="'$(Language)' == 'C#'">$(DefineConstants);DIAG_SUPPORTS_SUPPRESSION</DefineConstants>
```
Comment (sharwell) [inline src/Microsoft.CodeAnalysis.Testing/Directory.Build.props:39]: ❗ These can't change. Light-up/reflection must be used to support the scenario in cases where the APIs do not exist.

**Review by sharwell — CHANGES_REQUESTED** [review]

File: Roslyn-SDK.sln [hunk 2]
```diff
@@ -20,6 +20,10 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Roslyn.SyntaxVisualizer.Extension", "src\VisualStudio.Roslyn.SDK\SyntaxVisualizer\Roslyn.SyntaxVisualizer.Extension\Roslyn.SyntaxVisualizer.Extension.csproj", "{BF3477AA-22C9-4751-A4C0-01EF30728AE9}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Microsoft.CodeAnalysis.Testing", "Microsoft.CodeAnalysis.Testing", "{A3E08CE3-2358-4D36-875B-82C99AC61CD8}"
+	ProjectSection(SolutionItems) = preProject
+		src\Microsoft.CodeAnalysis.Testing\Directory.Build.props = src\Microsoft.CodeAnalysis.Testing\Directory.Build.props
+		src\Microsoft.CodeAnalysis.Testing\Directory.Build.targets = src\Microsoft.CodeAnalysis.Testing\Directory.Build.targets
+	EndProjectSection
```
Comment (sharwell) [inline Roslyn-SDK.sln:26]: 💡 These don't need to be added to the solution. They are visible in Solution Explorer with this button:

image

Comment (tom-englert) [reply Roslyn-SDK.sln:26]: It's not about viewing the file, it's about getting FindInFiles with scope solution working as expected. (for guys like me being not so used to all the features and tweaks in an unknown solution)
However I won't insist in adding this...

Comment (sharwell) [reply Roslyn-SDK.sln:26]: I find it's always a losing battle to keep solution files synchronized with the solution. The only way it's _always_ consistent is to omit them all and point users to the solution filter button. 😄

File: src/Microsoft.CodeAnalysis.Testing/Directory.Build.props (hunk 1 above)
Comment (tom-englert) [reply src/Microsoft.CodeAnalysis.Testing/Directory.Build.props:39]: OK, then I will do another PR using reflection.

Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Directory.Build.props:39]: We have a few places where it's already used. See `CodeActionExtensions` for an example:

https://github.com/dotnet/roslyn-sdk/blob/27fe3d6060150031b532cc7255259c40a37a5fda/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Extensions/CodeActionExtensions.cs#L15-L24

Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Directory.Build.props:39]: Also the support for AnalyzerConfigDocuments https://github.com/dotnet/roslyn-sdk/pull/734

**Comment by tom-englert** [comment]
Replaced by #917

**Pull request closed without merging**